lib/ecto/adapters/connection.ex

```diff
@@ -12,6 +12,39 @@ defmodule Ecto.Adapters.Connection do
 
   use Behaviour
 
+  defmacro __using__(_) do
+    quote do
+
+      @behaviour Ecto.Adapters.Connection
+
+      def after_connect(conn, opts) do
+        repo = opts[:repo]
+        if function_exported?(repo, :after_connect, 1) do
+          try do
+            Task.async(fn -> repo.after_connect(conn) end)
+            |> Task.await(opts[:timeout])
+          catch
+            :exit, {:timeout, [Task, :await, [%Task{pid: task_pid}, _]]} ->
+              Process.exit(task_pid, :kill)
+              {:error, :timeout}
+            :exit, {reason, {Task, :await, _}} ->
+              {:error, reason}
+            class, reason ->
```

Comment:
We don't need a catch all here because `after_connect` is carried out in the task and we only want to catch errors from the task, which will be of the form: `:exit, {reason, {Task, :await, [task, timeout]}}`.